tools/templatetest/main.go

```diff
@@ -0,0 +1,318 @@
+// Copyright 2024 MongoDB Inc
```

Comment:
@jeroenvervaeke currently we have the `templates_test.go` and also this tool, what is the relationship between the two? do you intend to keep both?

Reply:
Since this is code analisys to keep the code from breaking and you don't actually need the cli to run any commands I would be in favour of moving this test to code health (either on evergreen or github actions)

Reply:
I'm sorry, I should have deleted `tools/templatetest/main.go`.
I used that file to test.

Reply:
> Since this is code analisys to keep the code from breaking and you don't actually need the cli to run any commands I would be in favour of moving this test to code health (either on evergreen or github actions)

As in, move the e2e test to the tools directory and call it from evergreen/github actions?

Reply:
Right now it is not part of `code health` because I don't want this test to block people if it gives a false positive. `code health` is required on PRs